README.md

```diff
@@ -4,30 +4,135 @@
 
-- Output shape: `EmbeddingDimension`
 
-      
+## Prerequisite
 
-## Example:
+Document, Executor, and Flow are the three fundamental concepts in Jina.
 
-Here is an example usage of the clip encoder.
+- [**Document**](https://github.com/jina-ai/jina/blob/master/.github/2.0/cookbooks/Document.md) is the basic data type in Jina;
+- [**Executor**](https://github.com/jina-ai/jina/blob/master/.github/2.0/cookbooks/Executor.md) is how Jina processes Documents;
+- [**Flow**](https://github.com/jina-ai/jina/blob/master/.github/2.0/cookbooks/Flow.md) is how Jina streamlines and scales Executors.
 
-```python
-    def process_response(resp):
-        ...
+*Learn them all, nothing more, you are good to go.*
+
+
+## Usage
+
+### Via Jinahub
+
+1. Clone the repo and build the Jinahub image
+
+	```shell
+	git clone https://github.com/jina-ai/executor-clip-image.git
+	cd executor-clip-image
+	jina hub push .
+	```
+
+1. Use the image in your codes
+
+	```python
+	from jinahub.encoder.clip_image import ClipImageEncoder
+	from jina import Flow, Document
+	import numpy as np
+	
+	f = Flow().add(
+	        uses=jinahub+docker://xgdfljc:v1,
+	        volumes='/your_home_folder/.cache/clip:/root/.cache/clip')
+	
+	def check_emb(resp):
+	    for doc in resp.data.docs:
+	        if doc.emb:
+	            assert doc.emb.shape == (512,)
+	
+	with f:
+		f.post(
+		    on='/foo', 
+		    inputs=Document(np.random.random((224, 224, 3))), 
+		    on_done=check_emb)
+	```
+
+
+### Via Docker
+
+1. Clone the repo and build the docker image
+
+	```shell
+	git clone https://github.com/jina-ai/executor-clip-image.git
+	cd executor-clip-image
+	docker build -t jinahub-clip-image .
+	```
+
+1. Use `jinahub-clip-image` in your codes
+
+	```python
+	from jinahub.encoder.clip_image import ClipImageEncoder
+	from jina import Flow, Document
+	import numpy as np
+	
+	f = Flow().add(
+	        uses=docker://jinahub-clip-image:latest,
+	        volumes='/your_home_folder/.cache/clip:/root/.cache/clip')
+	
+	def check_emb(resp):
+	    for doc in resp.data.docs:
+	        if doc.emb:
+	            assert doc.emb.shape == (512,)
+	
+	with f:
+		f.post(
+		    on='/foo', 
+		    inputs=Document(np.random.random((224, 224, 3))), 
+		    on_done=check_emb)
+	```
+
+### Via Pypi
+
+1. Install the `jinahub-clip-image`
+
+	```bash
+	pip install git+https://github.com/jina-ai/executor-clip-image.git
+	```
+2. Use `jinahub-clip-image` in your code
+
+	```python
+	from jinahub.encoder.clip_image import ClipImageEncoder
+	from jina import Flow, Document
+	import numpy as np
+	
+	f = Flow().add(uses=ClipImageEncoder)
+	
+	def check_emb(resp):
+	    for doc in resp.data.docs:
+	        if doc.emb:
+	            assert doc.emb.shape == (512,)
+	
+	with f:
+		f.post(
+		    on='/foo', 
+		    inputs=Document(np.random.random((224, 224, 3))), 
+		    on_done=check_emb)
+	```
+
+
+
+## Parameters
```

Comment:
I think we need an exhaustive list

Reply:
hmmm, i think we need to find a way to include docstrings into README automatically.